src/plugins/kibana_usage_collection/server/collectors/cloud/detector/aws.test.ts

```diff
@@ -6,120 +6,120 @@
  * Side Public License, v 1.
  */
 
-import fs from 'fs';
-import type { Request, RequestOptions } from './cloud_service';
+/* eslint-disable dot-notation */
+jest.mock('node-fetch');
+jest.mock('fs/promises');
 import { AWSCloudService, AWSResponse } from './aws';
 
-type Callback = (err: unknown, res: unknown) => void;
-
-const AWS = new AWSCloudService();
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const fetchMock = require('node-fetch') as jest.Mock;
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const { readFile } = require('fs/promises') as { readFile: jest.Mock };
 
 describe('AWS', () => {
-  const expectedFilenames = ['/sys/hypervisor/uuid', '/sys/devices/virtual/dmi/id/product_uuid'];
-  const expectedEncoding = 'utf8';
-  // mixed case to ensure we check for ec2 after lowercasing
-  const ec2Uuid = 'eC2abcdef-ghijk\n';
-  const ec2FileSystem = {
-    readFile: (filename: string, encoding: string, callback: Callback) => {
-      expect(expectedFilenames).toContain(filename);
-      expect(encoding).toEqual(expectedEncoding);
-
-      callback(null, ec2Uuid);
-    },
-  } as typeof fs;
+  const mockIsWindows = jest.fn();
+  const awsService = new AWSCloudService();
+  awsService['_isWindows'] = mockIsWindows.mockReturnValue(false);
+  readFile.mockResolvedValue('eC2abcdef-ghijk\n');
 
+  beforeEach(() => jest.clearAllMocks());
   it('is named "aws"', () => {
-    expect(AWS.getName()).toEqual('aws');
+    expect(awsService.getName()).toEqual('aws');
   });
 
   describe('_checkIfService', () => {
     it('handles expected response', async () => {
       const id = 'abcdef';
-      const request = ((req: RequestOptions, callback: Callback) => {
-        expect(req.method).toEqual('GET');
-        expect(req.uri).toEqual(
-          'http://169.254.169.254/2016-09-02/dynamic/instance-identity/document'
-        );
-        expect(req.json).toEqual(true);
-
-        const body = `{"instanceId": "${id}","availabilityZone":"us-fake-2c", "imageId" : "ami-6df1e514"}`;
-
-        callback(null, { statusCode: 200, body });
-      }) as Request;
-      // ensure it does not use the fs to trump the body
-      const awsCheckedFileSystem = new AWSCloudService({
-        _fs: ec2FileSystem,
-        _isWindows: false,
+
+      fetchMock.mockResolvedValue({
+        json: () =>
+          `{"instanceId": "${id}","availabilityZone":"us-fake-2c", "imageId" : "ami-6df1e514"}`,
+        status: 200,
+        ok: true,
       });
 
-      const response = await awsCheckedFileSystem._checkIfService(request);
+      const response = await awsService['_checkIfService']();
+      expect(readFile).toBeCalledTimes(0);
+      expect(fetchMock).toBeCalledTimes(1);
```

Comment:
Is asserting against the number of times `fetchMock` is called not redundant considering that we assert against what the mock is called with?
In other words, are there specific situations that might cause `fetch` to be called multiple times?

Reply:
we have no way of detecting if `fetch` was called multiple times by any reason without this `expect`. Checking what the fetech is being called with wouldnt detect how many times the fetch was called. Having this check helps a lot when we're refactoring code in the future as it might catch unexpected changes.